File: 251004/Khanenko/Lab_4-5/publisher/module-publisher/module-publisher/src/main/java/bsuir/khanenko/modulepublisher/service/MessageService.java
package bsuir.khanenko.modulepublisher.service;

import bsuir.khanenko.modulepublisher.dto.MessageRequestTo;
import bsuir.khanenko.modulepublisher.dto.MessageResponseTo;
import bsuir.khanenko.modulepublisher.dto.MessageUpdate;

import java.util.List;
import java.util.Optional;

public interface MessageService {
    MessageResponseTo create(MessageRequestTo message);

    MessageResponseTo update(MessageUpdate updatedMessage);

    void deleteById(Long id);

    List<MessageResponseTo> findAll();

    Optional<MessageResponseTo> findById(Long id);
}

File: 251003/Golovin/appLab5/common/src/main/kotlin/com/example/eger/dc/lab5/service/MessageService.kt
package com.example.eger.dc.lab5.service

import com.example.eger.dc.lab5.dto.request.MessageRequestTo
import com.example.eger.dc.lab5.dto.request.MessageRequestToId
import com.example.eger.dc.lab5.dto.response.MessageResponseTo

interface MessageService {
	suspend fun create(requestTo: MessageRequestTo?): MessageResponseTo?

	suspend fun deleteById(id: Long): Boolean

	suspend fun getAll(): List<MessageResponseTo>

	suspend fun getById(id: Long): MessageResponseTo?

	suspend fun update(requestTo: MessageRequestToId?): MessageResponseTo?
}